x-pack/plugins/endpoint/server/routes/resolver/children.ts

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import _ from 'lodash';
+import { schema } from '@kbn/config-schema';
+import { RequestHandler, Logger } from 'kibana/server';
+import { extractEntityID } from './utils/normalize';
+import { getPaginationParams } from './utils/pagination';
+import { LifecycleQuery } from './queries/lifecycle';
+import { ChildrenQuery } from './queries/children';
+
+interface ChildrenQueryParams {
+  after?: string;
+  limit: number;
+  legacyEndpointID?: string;
+}
+
+interface ChildrenPathParams {
+  id: string;
+}
+
+export const validateChildren = {
+  params: schema.object({ id: schema.string() }),
+  query: schema.object({
+    after: schema.maybe(schema.string()),
+    limit: schema.number({ defaultValue: 10, min: 1, max: 100 }),
+    legacyEndpointID: schema.maybe(schema.string()),
+  }),
+};
+
+export function handleChildren(
+  log: Logger
+): RequestHandler<ChildrenPathParams, ChildrenQueryParams> {
+  return async (context, req, res) => {
+    const {
+      params: { id },
+      query: { limit, after, legacyEndpointID },
+    } = req;
+    try {
+      const pagination = getPaginationParams(limit, after);
+
+      const client = context.core.elasticsearch.dataClient;
+      const childrenQuery = new ChildrenQuery(legacyEndpointID, pagination);
+      const lifecycleQuery = new LifecycleQuery(legacyEndpointID);
+
+      // Retrieve the related child process events for a given process
+      const { total, results: events, next } = await childrenQuery.search(client, id);
```

Comment:
Do are all the `.search` calls handle ES errors properly or is this what the try is for?

Reply:
That's what the `try` is for, we will return a 500 to the UI I believe if there's an elasticsearch error. We do want to revisit some of that in a future PR because I would think we'd just want to return a 400 or something instead. I should probably make a ticket.

Reply:
Ticket: https://github.com/elastic/endpoint-app-team/issues/180